TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/qual/QualHardware.java

```diff
@@ -52,24 +67,40 @@ public class QualHardware {
     public DcMotorEx backRight;
     public DcMotorEx lift;
     //public Servo claw;
-    public BNO055IMU imu;
-    //public RI3WComputerVisionProcessor pipeline;
+    public double speed = .15;
+
+    // public BNO055IMU imu;
 
-    //public OpenCvCamera camera;
-    //public VisionPortal visionPortal;
-    //private RI3WComputerVisionProcessor visionProcessor;
+    RevHubOrientationOnRobot.UsbFacingDirection usbFacingDirection;
+    RevHubOrientationOnRobot.LogoFacingDirection logoDirection;
+    public IMU imu;
 
-    /*public void init(HardwareMap hardwareMap, Telemetry telemetry) {
-        init(hardwareMap, telemetry, RI3WComputerVisionProcessor.AllianceColor.BLUE);
+    public Servo dragger;
+    public void init(HardwareMap hardwareMap, Telemetry telemetry) {
+        init(hardwareMap, telemetry, ComputerVisionProcessor.AllianceColor.BLUE);
     }
-     */
-    public void init(HardwareMap hardwareMap, Telemetry telemetry){
+    public void init(HardwareMap hardwareMap, Telemetry telemetry, ComputerVisionProcessor.AllianceColor teamColor){
+
+        aprilTag = new AprilTagProcessor.Builder()
+                .setLensIntrinsics(FX, FY, CX, CY)
+                .setOutputUnits(DistanceUnit.INCH, AngleUnit.DEGREES)
+                .build();
+        propDetectionProcessor = new ComputerVisionProcessor(telemetry);
+        visionPortal = new VisionPortal.Builder()
+                .setCamera(hardwareMap.get(WebcamName.class, "Webcam 1"))
+                .addProcessors(aprilTag, propDetectionProcessor)
+                .setCameraResolution(new Size(1280, 720))
```

Comment:
Do you need 1280x720 resolution? It is also contributing to the memory usage. Perhaps a lower resolution would not cause the crash at all, if the garbage collection is able to keep up with it.